### PR TITLE
Update benchmark CI to run only on approved PRs and Swift changes

### PR DESCRIPTION
**Pull request by ptoffy**

### CODE REVIEW

**Review by gwynne — APPROVED** [review]

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 on:
   workflow_dispatch:
+  pull_request_review:
+    types: [submitted]
   pull_request:
     branches: [ main ]
-  
+    types: [synchronize]
+    paths:
+      - '**/*.swift'
```
Comment (MahdiBM) [inline .github/workflows/benchmark.yml:10]: add the CI file itself as well?

Comment (ptoffy) [reply .github/workflows/benchmark.yml:10]: Mhh why?

Comment (MahdiBM) [reply .github/workflows/benchmark.yml:10]: If we change the CI file we'd want to make sure it is still functional?

Comment (0xTim) [reply .github/workflows/benchmark.yml:10]: Yes this. We can also ignore the tests as well

**Review by 0xTim — APPROVED** [review]
As Mahdi says, we should update this to check for the benchmark files as well. But otherwise LGTM

**Pull request merged**